pkg/plugins/manager.go

```diff
@@ -0,0 +1,113 @@
+package plugins
+
+import (
+	"errors"
+	"os"
+	"os/exec"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-plugin"
+	"github.com/treeverse/lakefs/pkg/logging"
+)
+
+var (
+	ErrPluginOfWrongType    = errors.New("plugin of the wrong type")
+	ErrPluginNameNotFound   = errors.New("unknown plugin name")
+	ErrUninitializedManager = errors.New("uninitialized plugins manager")
+)
+
+var allowedProtocols = []plugin.Protocol{
+	plugin.ProtocolGRPC,
+}
+
+// PluginIdentity identifies the plugin's version and executable location.
+type PluginIdentity struct {
+	Version int
+	// Cmd is the command that is used to run the plugin executable on the local filesystem.
+	Cmd *exec.Cmd
+}
+
+// PluginHandshake includes handshake properties for the plugin.
```

Comment:
I wonder, isn't this tied 1:1 with PluginIdentity?  I know we've been over this before, I don't see in the code why they need to be separate.

Reply:
the previous argument was that a handshake is more of an auth entity rather than an identifier.